pkg/app/pipedv1/plugin/kubernetes/provider/kubectl_test.go

```diff
@@ -0,0 +1,343 @@
+metadata:
+  name: nginx-deployment
+  namespace: test
+  labels:
+    env: test
+spec:
+  selector:
+    matchLabels:
+      env: test
+  template:
+    metadata:
+      labels:
+        env: test
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-service
+  namespace: test
+  labels:
+    env: test
+spec:
+  selector:
+    env: test
+  ports:
+    - port: 80
+      targetPort: 8080
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-config
+  labels:
+    env: test
+data:
+  key: value
+`),
+			want: []ResourceKey{
+				{
+					groupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					namespace: "test",
+					name:      "nginx-deployment",
+				},
+				{
+					groupKind: schema.GroupKind{
+						Group: "",
+						Kind:  "Service",
+					},
+					namespace: "test",
+					name:      "nginx-service",
+				},
+				{
+					groupKind: schema.GroupKind{
+						Group: "",
+						Kind:  "ConfigMap",
+					},
+					namespace: "default",
+					name:      "test-config",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:      "get all namespace-scoped resources in a specific namespace",
+			namespace: "test",
+			selectors: []string{"env=test"},
+			manifests: mustParseManifests(t, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: test
+  labels:
+    env: test
+spec:
+  selector:
+    matchLabels:
+      env: test
+  template:
+    metadata:
+      labels:
+        env: test
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-service
+  namespace: test
+  labels:
+    env: test
+spec:
+  selector:
+    env: test
+  ports:
+    - port: 80
+      targetPort: 8080
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-config
+  labels:
+    env: test
+data:
+  key: value			
+`),
+			want: []ResourceKey{
+				{
+					groupKind: schema.GroupKind{
+						Group: "apps",
+						Kind:  "Deployment",
+					},
+					namespace: "test",
+					name:      "nginx-deployment",
+				},
+				{
+					groupKind: schema.GroupKind{
+						Group: "",
+						Kind:  "Service",
+					},
+					namespace: "test",
+					name:      "nginx-service",
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// prepare k8s cluster
+			restConfig := setupEnvTest(t)
+			kubeconfig, err := kubeconfigFromRestConfig(restConfig)
+			require.NoError(t, err)
+
+			kubeconfigPath := path.Join(t.TempDir(), "kubeconfig")
+			err = os.WriteFile(kubeconfigPath, []byte(kubeconfig), 0755)
+			require.NoError(t, err)
+
+			kubectl := NewKubectl(path.Join(os.Getenv("KUBEBUILDER_ASSETS"), "kubectl"))
+
+			// create namespace for testing
+			err = kubectl.CreateNamespace(t.Context(), kubeconfigPath, "test")
+			require.NoError(t, err)
```

Comment:
prepare the environment for each test case to avoid conflict over the name of each resource by executing in parallel.